src/hooks/useAllServices.ts

```diff
@@ -95,103 +100,114 @@ const filterAllServicesSections = (allServicesLinks: AllServicesSection[], filte
+        };
+      } else if (isAllServicesLink(item as AllServicesLink)) {
+        return item;
+      }
+      const [bundle, nav] = (item as string).split('.');
+      const currBundle = availableLinks.find(({ id }) => id === bundle)?.items || {};
+      return Object.values(currBundle).find(({ id }) => id === nav);
+    })
+    .filter(Boolean) as (AllServicesLink | AllServicesGroup)[];
+
 const useAllServices = () => {
-  const [{ availableLinks, ready, error }, setState] = useState<{
+  const [{ ready, error, availableSections }, setState] = useState<{
     error: boolean;
     ready: boolean;
-    availableLinks: NavItem[];
+    availableLinks: BundleNav[];
+    availableSections: AllServicesSection[];
   }>({
     ready: false,
     availableLinks: [],
     error: false,
+    availableSections: [],
   });
   const isMounted = useRef(false);
   const [filterValue, setFilterValue] = useState('');
   // TODO: move constant once the AppFilter is fully replaced
   const bundles = requiredBundles;
+  const fetchNavitation = useCallback(
+    () =>
+      Promise.all(
+        bundles.map((fragment) =>
+          axios
+            .get<BundleNav>(`${getChromeStaticPathname('navigation')}/${fragment}-navigation.json?ts=${Date.now()}`)
+            .then(handleBundleResponse)
+            .catch((err) => {
+              console.error('Unable to load appfilter bundle', err, fragment);
+              return [];
+            })
+        )
+      ),
+    []
+  );
+  const fetchSections = useCallback(
+    async (): Promise<AllServicesSection[]> => (await axios.get(`${getChromeStaticPathname('services')}/services.json?ts=${Date.now()}`)).data,
+    []
+  );
   useEffect(() => {
     isMounted.current = true;
-    Promise.all(
-      bundles.map((fragment) =>
-        axios
-          .get<BundleNavigation>(`${getChromeStaticPathname('navigation')}/${fragment}-navigation.json?ts=${Date.now()}`)
-          .then(handleBundleResponse)
-          .catch((err) => {
-            console.error('Unable to load appfilter bundle', err, fragment);
-            return [];
-          })
-      )
-    ).then((bundleItems) => {
+    const setNavigation = async () => {
+      const bundleItems = await fetchNavitation();
+      const sections = await fetchSections();
       if (isMounted.current) {
-        const availableLinks = parseBundlesToObject(bundleItems.flat());
+        const availableLinks = (bundleItems as BundleNav[]).map((bundle) => {
+          return {
+            ...bundle,
+            items: parseBundlesToObject(bundle.links.flat()),
+          };
+        });
+        const availableSections = sections
+          .reduce<AllServicesSection[]>((acc, { links, ...rest }) => {
+            return [
+              ...acc,
+              {
+                ...rest,
+                links: findNavItems((links || []) as string[], availableLinks).filter(Boolean),
```

Comment:
You can initialize the links as empty array in the argument and you won't require the type cast.

Reply:
Aaah, good point!